src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2534,7 +2546,104 @@ namespace winrt::TerminalApp::implementation
+        {
+            // Add the new tab to the list of our tabs.
+            auto newTabImpl = winrt::make_self<SettingsTab>();
+            _tabs.Append(*newTabImpl);
+
+            // Give the tab its index in the _tabs vector so it can manage its own SwitchToTab command.
+            newTabImpl->UpdateTabViewIndex(_tabs.Size() - 1);
+
+            // Don't capture a strong ref to the tab. If the tab is removed as this
+            // is called, we don't really care anymore about handling the event.
+            auto weakTab = make_weak(newTabImpl);
+
+            auto tabViewItem = newTabImpl->TabViewItem();
+            _tabView.TabItems().Append(tabViewItem);
+
+            // GH#6570
+            // The TabView does not apply compact sizing to items added after Compact is enabled.
+            // By forcibly reapplying compact sizing every time we add a new tab, we'll make sure
+            // that it works.
+            // Workaround from https://github.com/microsoft/microsoft-ui-xaml/issues/2711
+            if (_tabView.TabWidthMode() == MUX::Controls::TabViewWidthMode::Compact)
+            {
+                _tabView.UpdateLayout();
+                _tabView.TabWidthMode(MUX::Controls::TabViewWidthMode::Compact);
+            }
+
+            tabViewItem.PointerPressed({ this, &TerminalPage::_OnTabClick });
+
+            // When the tab is closed, remove it from our list of tabs.
+            newTabImpl->Closed([tabViewItem, weakThis{ get_weak() }](auto&& /*s*/, auto&& /*e*/) {
+                if (auto page{ weakThis.get() })
+                {
+                    page->_switchToSettingsCommand = nullptr;
+                    page->_RemoveOnCloseRoutine(tabViewItem, page);
+                }
+            });
+
+            _switchToSettingsCommand = newTabImpl->SwitchToTabCommand();
+
+            // This kicks off TabView::SelectionChanged, in response to which
+            // we'll attach the terminal's Xaml control to the Xaml root.
+            _tabView.SelectedItem(tabViewItem);
+        }
+        else
+        {
+            _actionDispatch->DoAction(_switchToSettingsCommand.Action());
+        }
+    }
+
+    // Method Description:
+    // - Returns a com_ptr to the implementation type of a TerminalTab at the given index.
+    //   If the tab is not a TerminalTab, returns nullptr.
+    // Arguments:
+    // - index: an unsigned integer index to a tab in _tabs
+    // Return Value:
+    // - If the tab is a TerminalTab, a com_ptr to the implementation type.
+    //   If the tab is not a TerminalTab, nullptr
+    winrt::com_ptr<TerminalTab> TerminalPage::_GetTerminalTabImpl(const uint32_t index) const
+    {
+        if (auto tab = _tabs.GetAt(index).try_as<TerminalApp::TerminalTab>())
+        {
+            winrt::com_ptr<TerminalTab> tabImpl;
+            tabImpl.copy_from(winrt::get_self<TerminalTab>(tab));
+            return tabImpl;
+        }
+        else
+        {
+            return nullptr;
+        }
```

Comment:
Would this work?
```suggestion
        return _GetTerminalTabImpl(_tabs.GetAt(index));
```

Or could we remove this helper function altogether and just use the one below instead?

Reply:
o good call, it's a one liner so I'll just replace this function call with that